.circleci/config.yml

```diff
@@ -207,7 +207,17 @@ commands:
   reinstall-go:
     steps:
       - run: sudo rm -rf /usr/local/go # Remove system go.
-      - run: curl https://dl.google.com/go/go1.18.linux-amd64.tar.gz -o /tmp/go.linux-amd64.tar.gz
+      - run: |
+          echo '5e05400e4c79ef5394424c0eff5b9141cb782da25f64f79d54c98af0a37f8d49 /tmp/go.linux-amd64.tar.gz' >/tmp/hash.txt
+          for _ in {1..3}; do
```

Comment:
You had a rare opportunity to use a do/until loop here, and missed it.  I'm so very disappointed. ;)

counter=3; until sha; do if $(( counter-- )); then echo fail; exit 1; curl; done